src/main/java/org/betonquest/betonquest/events/VariableEvent.java

```diff
@@ -29,8 +30,11 @@ public VariableEvent(final Instruction instruction) throws InstructionParseExcep
         objectiveID = instruction.getObjective();
         key = instruction.next();
         keyVariables = BetonQuest.resolveVariables(key);
-        value = instruction.next();
-        valueVariables = BetonQuest.resolveVariables(value);
+        final List<String> args = new ArrayList<>(instruction.size());
+        while(instruction.hasNext()) args.add(instruction.next());
```

Comment:
This will include any potential defined condition for this event ([see here](https://betonquest.org/2.0/Documentation/Scripting/About-Scripting/#events)). This mistake is not unique to this implementation but I don't have a easy to use solution for that now (the only "solution" I know is inside the [notify event](https://github.com/BetonQuest/BetonQuest/blob/main/src/main/java/org/betonquest/betonquest/quest/event/notify/NotifyEventFactory.java).

Reply:
Oh, I completely forgot about it. I'll try to rewrite its parser to handle it better